sdk/synapse/synapse-spark/review/synapse-spark.api.md

```diff
@@ -0,0 +1,525 @@
+## API Report File for "@azure/synapse-spark"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import * as coreHttp from '@azure/core-http';
+import { HttpResponse } from '@azure/core-http';
+import { OperationOptions } from '@azure/core-http';
+import { PipelineOptions } from '@azure/core-http';
+import { TokenCredential } from '@azure/core-http';
+
+// @public (undocumented)
+export type CancelSparkBatchJobOptions = OperationOptions;
+
+// @public (undocumented)
+export type CancelSparkSessionOptions = OperationOptions;
+
+// @public (undocumented)
+export type CancelSparkStatementOptions = OperationOptions;
+
+// @public
+export interface CreateSparkBatchJobOptions extends coreHttp.OperationOptions {
+    detailed?: boolean;
+}
+
+// @public
+export type CreateSparkBatchJobResponse = SparkBatchJob & {
```

Comment:
lately we've found it easier to use a type that attaches `_response` to the return shape (e.g. https://github.com/Azure/azure-sdk-for-js/blob/4726c352e06b21c3409726defc6cb991841e91a7/sdk/servicebus/service-bus/src/serviceBusAtomManagementClient.ts#L99)

So you can have return values like `WithResponse<SparkBatchJob>` and avoid needing to declare this extra type

Reply:
Cool, I will try it. Do we need stay the same structure with generated code? like:
```
export type WithResponse<T extends object> = T & {
  _response: HttpResponse & {
    bodyAsText: string;
    parsedBody: T;
  };
};
```
[https://github.com/Azure/azure-sdk-for-js/blob/1f8a171a648dcd53de6ebd4b8e5d420fa9c54abf/sdk/synapse/synapse-spark/src/generated/models/index.ts#L391](https://github.com/Azure/azure-sdk-for-js/blob/1f8a171a648dcd53de6ebd4b8e5d420fa9c54abf/sdk/synapse/synapse-spark/src/generated/models/index.ts#L391)

---
In reply to: [521605528](https://github.com/Azure/azure-sdk-for-js/pull/12487#discussion_r521605528) [](ancestors = 521605528)

Reply:
I think as long as you have `HttpResponse`, that'll be enough. 

`parsedBody` and `bodyAsText` aren't particularly useful to consumers -- they are internal details that usually aren't meaningful for debugging.